surface/secretsmanager/management/commands/import_git_secrets.py

```diff
@@ -0,0 +1,217 @@
+                        author = subprocess.run(
+                            f"git -C {repo_path} log -1 --format=%ae {commit} -- {file_path}",
+                            stdout=subprocess.PIPE,
+                            shell=True,
+                            text=True
+                        ).stdout.strip()
+                        
+                        date = subprocess.run(
+                            f"git -C {repo_path} log -1 --format=%aI {commit} -- {file_path}",
+                            stdout=subprocess.PIPE,
+                            shell=True,
+                            text=True
+                        ).stdout.strip()
+
+                        yield {
+                            'commit': commit,
+                            'file': file_path,
+                            'author': author,
+                            'date': date
+                        }
+
+        except subprocess.CalledProcessError as e:
+            self.stdout.write(self.style.ERROR(f"Error scanning git history: {e}"))
+
+    def calculate_file_hash(self, repo_path, commit_hash, file_path):
+        """Calculate SHA256 hash of file content at specific commit"""
+        try:
+            result = subprocess.run(
+                f"git -C {repo_path} show {commit_hash}:{file_path}",
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                shell=True,
+                check=True
+            )
+            content = result.stdout
+            return hashlib.sha256(content).hexdigest()
+        except subprocess.CalledProcessError as e:
+            self.stdout.write(self.style.ERROR(f"Error reading file {file_path}: {e}"))
+            return None
+
+    def handle(self, *args, **options):
+        repo_path = options['repo_path']
+        self.org = options['org']
+
+        self.stdout.write(f"Scanning repository: {repo_path}")
+        
+        repo_name = os.path.basename(repo_path)
+        repo_url = f"https://github.com/{self.org}/{repo_name}"
+        default_branch = self.get_default_branch(repo_path)
+        
+        git_source, _ = GitSource.objects.get_or_create(
+            repo_url=repo_url,
+            defaults={'branch': default_branch}
+        )
+
+        # Track files by their hash
+        secret_sources = {}
+        
+        # First, collect all instances of each secret
+        for file_info in self.scan_git_history(repo_path):
+            file_hash = self.calculate_file_hash(
+                repo_path, 
+                file_info['commit'], 
+                file_info['file']
+            )
+            
+            if not file_hash:
+                continue
+
+            source_data = {
+                "commit": file_info['commit'],
+                "file": file_info['file'],
+                "email": file_info['author'],
+                "repository": repo_url,
+                "timestamp": file_info['date'],
+                "line": "1"
```

Comment:
1 should be a default only when there's no real line from the file. As it stands, everything here will be line 1, which is non-ideal.